<?xml version="1.0"?>
<launch>

	<!-- arguments -->
	<!-- <arg name="domain_path"	default="/home/maciek/dyplom_ws/src/dyplom/pddl/domain.pddl" />
	<arg name="problem_path"	default="/home/maciek/dyplom_ws/src/dyplom/pddl/problem.pddl" />
    <arg name="gen_problem_path"	default="/home/maciek/dyplom_ws/src/dyplom/pddl/gen_problem.pddl" /> -->

	<arg name="domain_path"	default="/home/maciek/dyplom_ws/src/dyplom/pddl/domain_simple.pddl" />
	<arg name="problem_path"	default="/home/maciek/dyplom_ws/src/dyplom/pddl/problem_simple.pddl" />
    <arg name="gen_problem_path"	default="/home/maciek/dyplom_ws/src/dyplom/pddl/gen_problem_simple.pddl" />

	<!-- <arg name="domain_path" value="$(find rosplan_turtlebot3_demo)/common/domain_turtlebot_demo.pddl" />
    <arg name="problem_path" value="" />
	<arg name="gen_problem_path" value="$(find rosplan_turtlebot3_demo)/common/problem.pddl" />
    <arg name="data_path" value="$(find rosplan_turtlebot3_demo)/common/" /> -->
	
	<arg name="data_path"	default="/home/maciek/dyplom_ws/src/dyplom/pddl/" />


	<!-- knowledge base -->
	<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
		<param name="domain_path" value="$(arg domain_path)" />
		<param name="problem_path" value="$(arg problem_path)" />
		<!-- conditional planning flags -->
		<param name="use_unknowns" value="false" />
	</node>

	<!-- problem generation -->
	<include file="$(find rosplan_planning_system)/launch/includes/problem_interface.launch">
		<arg name="knowledge_base"   value="rosplan_knowledge_base" />
		<arg name="domain_path"      value="$(arg domain_path)" />
		<arg name="problem_path"     value="$(arg gen_problem_path)" />
		<arg name="problem_topic"    value="problem_instance" />
	</include>

    <!-- planner interface -->
	<include file="$(find rosplan_planning_system)/launch/includes/planner_interface.launch">
		<arg name="use_problem_topic"    value="true" />
		<arg name="problem_topic"        value="/rosplan_problem_interface/problem_instance" />
		<arg name="planner_topic"        value="planner_output" />
		<arg name="domain_path"          value="$(arg domain_path)" />
		<arg name="problem_path"         value="$(arg gen_problem_path)" />
		<arg name="data_path"            value="$(arg data_path)" />
		<arg name="planner_command"      value="timeout 10 $(find rosplan_planning_system)/common/bin/popf DOMAIN PROBLEM" />
	</include>

	<!-- plan parsing -->
	<node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
		<param name="knowledge_base" value="rosplan_knowledge_base" />
		<param name="planner_topic"  value="/rosplan_planner_interface/planner_output" />
		<param name="plan_topic"     value="complete_plan" />
	</node>

	<!-- plan dispatching -->
	<node name="rosplan_plan_dispatcher" pkg="rosplan_planning_system" type="pddl_simple_plan_dispatcher" respawn="false" output="screen">
		<param name="knowledge_base"        value="rosplan_knowledge_base" />
		<param name="plan_topic"            value="/rosplan_parsing_interface/complete_plan" />
		<param name="action_dispatch_topic" value="action_dispatch" />
		<param name="action_feedback_topic" value="action_feedback" />
	</node>
</launch>
